src/status_im/contexts/onboarding/events.cljs

```diff
@@ -184,27 +183,35 @@
  (fn [{:keys [db]} [auth-method]]
    {:db (assoc db :auth-method auth-method)}))
 
-(rf/defn onboarding-new-account-finalize-setup
-  {:events [:onboarding/finalize-setup]}
-  [{:keys [db]}]
-  (let [masked-password    (get-in db [:onboarding/profile :password])
-        key-uid            (get-in db [:profile/profile :key-uid])
-        syncing?           (get-in db [:onboarding/profile :syncing?])
-        auth-method        (get-in db [:onboarding/profile :auth-method])
-        biometric-enabled? (= auth-method constants/auth-method-biometric)]
-    (cond-> {:db (assoc db :onboarding/generated-keys? true)}
-      biometric-enabled?
-      (assoc :keychain/save-password-and-auth-method
-             {:key-uid         key-uid
-              :masked-password (if syncing?
-                                 masked-password
-                                 (security/hash-masked-password masked-password))
-              :on-success      (fn []
-                                 (rf/dispatch [:onboarding/set-auth-method auth-method])
-                                 (when syncing?
-                                   (rf/dispatch
-                                    [:onboarding/navigate-to-enable-notifications-from-syncing])))
-              :on-error        #(log/error "failed to save biometrics"
-                                           {:key-uid key-uid
-                                            :error   %})}))))
+(rf/reg-event-fx
+ :onboarding/set-reset-display-name
+ (fn [{:keys [db]} [reset-display-name?]]
+   (let [initial-profile-data {:reset-display-name? reset-display-name?
+                               :display-name        (if reset-display-name? "temp username" "")}]
+     {:db (assoc db :onboarding/profile initial-profile-data)})))
+
+(rf/reg-event-fx
+ :onboarding/finalize-setup
+ (fn [{db :db}]
+   (let [{:keys [password syncing? auth-method
+                 reset-display-name?]} (:onboarding/profile db)
+         {:keys [key-uid] :as profile} (:profile/profile db)
+         biometric-enabled?            (= auth-method constants/auth-method-biometric)]
+     {:db (assoc db :onboarding/generated-keys? true)
+      :fx [(when reset-display-name?
+             [:dispatch [:profile/set-default-profile-name profile]])
```

Comment:
This call updates the profile name, it's called when finalizing the onboarding set-up and the user is already logged in